core/db/migrate/20230328042835_improve_strictness_of_spree_line_items.rb

```diff
@@ -0,0 +1,42 @@
+class ImproveStrictnessOfSpreeLineItems < ActiveRecord::Migration[7.0]
```

Comment:
```suggestion
class ImproveStrictnessOfSpreeLineItems < ActiveRecord::Migration[5.2]
```

Reply:
☝️ this will likely make the test pass on all Rails versions.

Reply:
I know that we do this in all our migrations, but I want to suggest that we do something like this

```rb
class ImproveStrictnessOfSpreeLineItems < ActiveRecord::Migration[ActiveRecord::Migration.current_version]
```

to make sure that new Stores get all features (like `t.bigint`, `NOT NULL` on timestamps, etc.) for later Rails versions.

To make it easier we could introduce a `Spree::Migration` base class.

Reply:
The only problem with this approach is that we won't be sure of how the database of users looks like anymore. A middle-ground could be using the last version of Rails that we support (currently 6.0) for new migrations?

Reply:
I mean, but correct me if I'm wrong, we won't know which migrations have been executed on different stores, depending on the version of Rails they are using. This way, we could have a future migration (or application code) that will need conditional paths depending on the Rails version used when the migration has been executed.

Reply:
We could just do a large migration and set all the timestamps as `not null`. That seems to be the appropriate solution. Then we could do another migration later where we make all the primary/foreign keys bigints.

Reply:
This should be fine after merging https://github.com/solidusio/solidus/pull/5012